[feat] Implement PWA

CHAERYEON-LEE opened the pull request:
### 1️⃣ PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [x] 의존성, 환경 변수, 빌드 관련 코드 업데이트
<br>

### 2️⃣ 이슈 번호
ex) close #24 
<br>
<br>

### 3️⃣ 변경 사항
PWA 구현(초기단계)
- npm next-pwa 모듈 install
- config 파일에 withPWA 모듈 추가
- manifest.json 작성
- 192x192, 512x512 크기 로고 추가
- _document.tsx에 meta 태그 추가

- .gitignore 에 PWA 관련 파일 추가
- build시  Dialog export 안됐다는 문제 해결
<br>
<br>

CHAERYEON-LEE committed:
feat: Implement PWA

PWA 구현(초기단계)
- npm next-pwa 모듈 install
- config 파일에 withPWA 모듈 추가
- manifest.json 작성
- 192x192, 512x512 크기 로고 추가
- _document.tsx에 meta 태그 추가

CHAERYEON-LEE committed:
style: Add .gitignore

- .gitignore 에 PWA 관련 파일 추가
- build시  Dialog export 안됐다는 문제 해결

pereng11 commented on public/manifest.json:
```diff
@@ -0,0 +1,24 @@
+{
+    "name": "mybuddyhero",
+    "short_name": "mybuddyhero",
```
이 속성은 어플리케이션 아이콘 아래 이름으로 표시된다고 합니다
대문자 + 띄어쓰기 하면 더 좋을 것 같아요

pereng11 commented on public/manifest.json:
```diff
@@ -0,0 +1,24 @@
+    "display": "standalone",
+    "theme_color": "#FFFFFF",
+    "background_color": "#FFFFFF",
+    "orientation": "portrait",
```
기기 방향을 결정하는 속성인데, 저희는 가로 방향이니까 'landscape'가 맞을 것 같습니다!

pereng11 reviewed (commented):
이 사이트 자료 참고했습니다!
https://ux.stories.pe.kr/225

CHAERYEON-LEE committed:
style: Add logo img file

PWA 위한 로고 이미지 파일 추가

CHAERYEON-LEE committed:
feat: Complete test PWA tablet

PWA 태블릿 테스트 완료
- .gitignore에 PWA file 추가했던부분 제거
- manifest favicon 추가
- _document.tsx 에 meta 태그 추가

pereng11 reviewed (approved):
PWA마스터 미쳤다 👍 👍
